providers/os/connection/docker/file.go

```diff
@@ -106,8 +106,11 @@ func (f *File) Readdirnames(n int) ([]string, error) {
 	}
 
 	// extract names
-	basenames := make([]string, len(directories))
+	basenames := []string{}
 	for i := range directories {
+		if directories[i] == "" {
+			continue
+		}
 		basenames[i] = filepath.Base(directories[i])
```

Comment:
Doesn't this leave the array with empty entries (`""`)?

Let's assume we skip `i == 2`. Then, the `basenames` array wouldn't have an entry there.
Perhaps it works, because we also do not call `filepath.Base` on this...

Reply:
The array isn’t a initialized with the size of the directories. That’s why it doesn’t contain empty entries